apps/src/blockly/addons/cdoXml.js

```diff
@@ -74,6 +75,29 @@ export default function initializeBlocklyXml(blocklyWrapper) {
   blocklyWrapper.Xml.createBlockOrderMap = createBlockOrderMap;
 }
 
+/**
+ * Adds a mutation element to a block if it should have an open miniflyout.
+ * CDO Blockly uses an unsupported method for serializing miniflyout state
+ * where arbitrary block attribute could be used to manage extra state.
+ * Mainline Blockly expects a mutator. The precence of the mutation element
+ * will trigger the block's domToMutation function to run, if it exists.
+ *
+ * @param {Element} blockElement - The XML element for a single block.
+ * @returns {Map} A map with partitioned block index as key and original index in the XML as value.
+ */
+export function addMutationToMinitoolboxBlocks(blockElement) {
```

Comment:
I don't think so, but it's still totally reasonable for us to check for the presence of the mutation before adding one. We can also check for the attribute before setting it. I'll update tomorrow!

Reply:
Updated. We know only add the mutation if it doesn't exist. We also remove the original block attribute so that the function will hit an early return if it happens to run twice on the same block.